cns/service/main.go

```diff
@@ -316,6 +377,22 @@ func main() {
 		go httpRestService.SendNCSnapShotPeriodically(cnsconfig.TelemetrySettings.SnapshotIntervalInMins, stopSnapshots)
 	}
 
+	// If CNS is running on managed DNC mode
+	if privateEndpoint != "" && infravnet != "" && nodeID != "" {
```

Comment:
See comment above. How can we know if CNS is running for mdnc mode? We need to validate this and throw error